tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -52,17 +57,32 @@ def get_storage_account_connection_string(id):
         return connection_string
 
     @staticmethod
-    def deploy_dedicated_function_app(parameters):
+    def is_consumption_plan(function_params):
+        return StringUtils.equal(function_params.service_plan['sku_tier'], 'dynamic')
+
+    @staticmethod
+    def deploy_function_app(parameters):
         function_app_unit = FunctionAppDeploymentUnit()
         function_app_params = \
             {'name': parameters.function_app_name,
              'resource_group_name': parameters.function_app_resource_group_name}
         function_app = function_app_unit.get(function_app_params)
+
         if function_app:
+            # retrieve the type of app service plan hosting the existing function app
+            session = local_session(Session)
```

Comment:
Was this thing session agnostic before this update? Maybe it doesn't matter - I see when it is used in Mailer this should still work.

Reply:
btw, we do need to test a mailer functions deployment before we merge this....

Reply:
It worked 2-3 days ago when I tested it.